src/components/visualizations/common/utils/test/drilldownEventing.spec.tsx

```diff
@@ -0,0 +1,365 @@
+        const fn = getClickableElementNameByChartType;
+        expect(fn(VisualizationTypes.LINE)).toBe('point');
+        expect(fn(VisualizationTypes.COLUMN)).toBe('bar');
+        expect(fn(VisualizationTypes.BAR)).toBe('bar');
+        expect(fn(VisualizationTypes.PIE)).toBe('slice');
+        expect(fn(VisualizationTypes.TABLE)).toBe('cell');
+        expect(() => {
+            fn('headline'); // headline is not defined
+        }).toThrowError();
+    });
+
+    it('should call default fire event on point click and fire correct data', () => {
+        const afm = { is: 'AFM' };
+        const drillConfig = { afm, onFiredDrillEvent: () => true };
+        const target = { dispatchEvent: jest.fn() };
+
+        chartClick(drillConfig, pointClickEventData as IHCDrillEventPoint,
+            target as any as EventTarget, VisualizationTypes.LINE);
+
+        jest.runAllTimers();
+
+        expect(target.dispatchEvent).toHaveBeenCalled();
+
+        expect(target.dispatchEvent.mock.calls[0][0].detail).toEqual({
+            executionContext: { is: 'AFM' },
+            drillContext: {
+                type: 'line',
+                element: 'point',
+                x: 1,
+                y: 2,
+                intersection: [
+                    {
+                        id: 'id',
+                        title: 'title',
+                        header: {
+                            identifier: 'identifier1',
+                            uri: 'uri1'
+                        }
+                    },
+                    {
+                        id: 'id',
+                        title: 'title',
+                        header: {
+                            identifier: 'identifier2',
+                            uri: 'uri2'
+                        }
+                    },
+                    {
+                        id: 'id',
+                        title: 'title',
+                        header: {
+                            identifier: 'identifier3',
+                            uri: 'uri3'
+                        }
+                    }
+                ]
+            }
+        });
+    });
+
+    it('should call user defined callback on point click', () => {
+        const afm = { is: 'AFM' };
+        const drillConfig = { afm, onFiredDrillEvent: jest.fn() };
+        const target = { dispatchEvent: () => true };
+
+        chartClick(drillConfig, pointClickEventData, target as any as EventTarget, VisualizationTypes.LINE);
+
+        jest.runAllTimers();
+
+        expect(drillConfig.onFiredDrillEvent).toHaveBeenCalled();
+    });
+
+    it('should call both default fire event and user defined callback on point click', () => {
+        const afm = { is: 'AFM' };
+        const drillConfig = { afm, onFiredDrillEvent: jest.fn() };
+        const target = { dispatchEvent: jest.fn() };
+
+        chartClick(drillConfig, pointClickEventData, target as any as EventTarget, VisualizationTypes.LINE);
```

Comment:
`target as any as EventTarget`  ???

Reply:
consider using `Partial<T>`

Reply:
I can do somethink like:

```javascript
export function fireDrillEvent(drillEventFunction: (data?: IDrillEvent) => boolean | void,
                               drillEventData: IDrillEvent,
                               target: Partial<EventTarget>) {
    const shouldDispatchPostMessage = drillEventFunction && drillEventFunction(drillEventData);

    if (shouldDispatchPostMessage !== false) {
        target.dispatchEvent(new CustomEvent('drill', {
            detail: drillEventData,
            bubbles: true
        }));
    }
}
```
And call it like 
```javascript
const eventHandler = jest.fn();
const target = {
     dispatchEvent: eventHandler
};
fireDrillEvent(undefined, eventData as IDrillEvent, target);
```

But I think it it not good approach since by this I say "every property of `EventTarget` is optional". This will shut up validator, but nothing says that `target` needs to have method `dispatchEvent` so its pretty close to use `any` type.

Reply:
ok I don't have strong opinion. 

only this
`as any as EventTarget` looks more weird then `Partial<EventTarget>`